### ISSUE TITLE
Train using custom dataset

### ISSUE THREAD

**Issue by JaromAktify** [malllabiisc/SGCP#7]
Hello - I'm trying to train using a custom dataset, but received the following error:
main.py: error: argument -dataset: invalid choice: 'my_data' (choose from 'QQPPos', 'ParaNMT50m')

Here's the command I ran after following all steps as outlined in the "custom data processing" section of https://github.com/malllabiisc/SGCP 
`$ python -m src.main -mode train -run_name my_data01 -dataset my_data -gpu 0 -bpe`

I expected a successful training run (i.e. training with QQPPos worked for me), but got the following when attempting to train on "my_data".  
_usage: main.py [-h] -mode {train,decode}
               [-datatype {test,trainrep,trainrep500}] [-len_sort]
               [-dataset {QQPPos,ParaNMT50m}] [-display_freq DISPLAY_FREQ]
               [-config_file CONFIG_FILE] [-project_name PROJECT_NAME]
               [-debug]
               [-pretrained_encoder {bert_cls,bert_avg,bert_max,bert_all,infersent,use}]
               [-cov_after_ep COV_AFTER_EP] [-run_name RUN_NAME] -gpu GPU
               [-seed SEED] [-log_fmt LOG_FMT] [-date_fmt DATE_FMT]
               [-ckpt_file CKPT_FILE] [-vocab_size VOCAB_SIZE] [-save_res]
               [-res_file RES_FILE] [-res_folder RES_FOLDER] [-single_side]
               [-height_dec HEIGHT_DEC] [-use_ptr] [-cov_wt COV_WT]
               [-cell_type CELL_TYPE] [-use_attn] [-attn_type ATTN_TYPE]
               [-hidden_size HIDDEN_SIZE] [-depth DEPTH]
               [-emb_size {50,100,200,300}] [-max_length MAX_LENGTH]
               [-s2sdprate S2SDPRATE] [-init_range INIT_RANGE]
               [-tree_height TREE_HEIGHT] [-lambda_dec_eop LAMBDA_DEC_EOP]
               [-lr_enc LR_ENC] [-lr_tree LR_TREE] [-lr_rnndec LR_RNNDEC]
               [-max_grad_norm MAX_GRAD_NORM] [-batch_size BATCH_SIZE]
               [-max_epochs MAX_EPOCHS] [-load_from_ep LOAD_FROM_EP]
               [-opt {adam,adadelta,sgd,asgd}] [-tfr TFR] [-bidirectional]
               [-dynamic_tree] [-bpe] [-use_word2vec] [-use_glove]
               [-beam_width BEAM_WIDTH] [-tree_height2 TREE_HEIGHT2]
               [-gcn_dim GCN_DIM] [-gcn_layers GCN_LAYERS]
               [-max_labels MAX_LABELS] [-max_nodes MAX_NODES] [-gcn_gating]
               [-no_gcn_gating]
main.py: error: argument -dataset: invalid choice: 'my_data' (choose from 'QQPPos', 'ParaNMT50m')_

Are there additional steps I can take to include custom datasets in addition to the QQPPos and ParaNMT50m options?

Thanks!

**Comment by ashutoshml** [comment]
@JaromAktify 

In the code, please modify the following line:
https://github.com/malllabiisc/SGCP/blob/ee2654889db48273f39c3c08d14a0b787ba48bd9/src/args.py#L11 
to  
`parser.add_argument('-dataset', type=str, default='QQPPos', choices=['QQPPos', 'ParaNMT50m', 'my_data'], help='Dataset to use')`


You may alternately use:
`parser.add_argument('-dataset', type=str, default='QQPPos', help='Dataset to use')`

**Comment by JaromAktify** [comment]
@ashutoshml 
Ashutosh, thank you for the help.  This worked!
